src/Analyser/MutatingScope.php

```diff
@@ -1969,16 +2150,25 @@ private function resolveType(Expr $node): Type
 						new ConstantStringType(';'),
 					]);
 				}
-				if ($resolvedConstantName === 'PHP_EOL') {
-					return new UnionType([
-						new ConstantStringType("\n"),
-						new ConstantStringType("\r\n"),
+				// iconv, https://www.php.net/manual/en/iconv.constants.php
+				if ($resolvedConstantName === 'ICONV_IMPL') {
+					return new IntersectionType([
+						new StringType(),
+						new AccessoryNonEmptyStringType(),
 					]);
 				}
-				if ($resolvedConstantName === '__COMPILER_HALT_OFFSET__') {
-					return new IntegerType();
+				// libxml, https://www.php.net/manual/en/libxml.constants.php
+				if ($resolvedConstantName === 'LIBXML_VERSION') {
+					return IntegerRangeType::fromInterval(1, null);
 				}
-				if ($resolvedConstantName === 'PHP_INT_MAX') {
+				if ($resolvedConstantName === 'LIBXML_DOTTED_VERSION') {
+					return new IntersectionType([
+						new StringType(),
+						new AccessoryNonEmptyStringType(),
+					]);
+				}
+				// openssl, https://www.php.net/manual/en/openssl.constants.php
+				if ($resolvedConstantName === 'OPENSSL_VERSION_NUMBER') {
 					return IntegerRangeType::fromInterval(1, null);
 				}
 
```

Comment:
another idea which comes to mind: would adding magic-constant types for e.g. `__FILE__` etc. be helpfull?

see https://www.php.net/manual/en/language.constants.magic.php

Reply:
I think it's a good idea, I wanted to add all the remaining useful constants anyways, I'll check later and just add them in this PR then.

Reply:
feel free to work in separate PRs.. I am just thinking out loud

Reply:
I tried to add them to the tests and found out that phpstan resolves them correctly already. Based on their usage, e.g. `__LINE__` would resolve to the line in the test file, `__FUNCTION__` would be empty as it's not inside a function and so on. I think adding them might make not much sense